messages/transaction.go

```diff
@@ -0,0 +1,28 @@
+package messages
+
+// TransactionType defines types of debits.
+type TransactionType string
+
+const (
+	// Deposit is a deposit transaction type.
+	Deposit TransactionType = "deposit"
+
+	// Withdrawal is a withdrawal transaction type.
+	Withdrawal TransactionType = "withdrawal"
+
+	// Transfer is a transfer transaction type.
+	Transfer TransactionType = "transfer"
+)
+
+// DebitFailureReason defines reasons why a debits may fail.
+type DebitFailureReason string
+
+const (
+	// InsufficientFunds means there was not enough funds available in the
+	// account to perform the debit.
+	InsufficientFunds DebitFailureReason = "insufficent-funds"
```

Comment:
Is there any particular reason we need to hyphenate these strings as opposed leave them as normal strings with spaces (as a phrase)?
IMHO, these string values are very similar to `go` error values that are determined by a regular string comparison.

I might even go a bit further and say why not make them proper `go` errors? I know, I know, these are not supposed to be programming errors, they are more like an expression of unhappy ending of the business process. But still, would not they look more appropriate as errors? E.g.
```
var InsufficientFunds = errors.New("insufficient funds")
```
... and then on the message they can be represented as errors, e.g.:
```
type WithdrawalDeclined struct {
	TransactionID string
	AccountID     string
	Amount        int64
	Reason        error
}
```
Hopefully everything I said makes sense in some sort of way...

Reply:
If not `go` errors can we make these strings similar to `go` representation of proto enums? so they are still represented by `int` but backed up by human readable strings?

Reply:
I think they are not really error values, especially not `go` specific error values. They are language independant values in messages. They could be routed to any kind of handler in any language. An `int` could be used though. I think I discussed this with @jmalloc, but I can't remember exactly why we decided on `string`'s.

@jmalloc , thoughts on this?

Reply:
Yeah, they are an enumeration with string values designed to be switched upon, not a human-readable message.  If we make them plain English then they appear as though they can be reworded, but doing so would break comparison against older events in an event-sourcing environment.

Using an integer is possible but is a bit cryptic for the purposes of the example. These values will be serialized as JSON when the example is used with Verity and so a string also makes it clear what the value means in that context.

Reply:
An int (error code style) would make it clear that its not something that should be changed, but yes that does make it more cryptic.

Reply:
cool, It make sense then to leave everything as is. Sorry for this deviation, guys.